pkg/promtail/targets/awss3target.go

```diff
@@ -0,0 +1,228 @@
+package targets
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/aws"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/loki/pkg/promtail/api"
+	"github.com/grafana/loki/pkg/promtail/positions"
+	"github.com/grafana/loki/pkg/promtail/scrape"
+)
+
+var (
+	s3ReadBytes = promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
I think we should avoid using `path` for these metrics. We're creating metrics in accordance with our stream cardinality here.

Reply:
`path` here means `object` name. What do you suggest here? Just consider the bucker name instead of each individual object?